tools/DatasetExtraction/dataset.py

```diff
@@ -0,0 +1,37 @@
+from abc import ABCMeta, abstractmethod
+from accessify import private, protected
+import chord
+
+class Dataset:
+    #i - song, j - chord
+    songlist = [[]]
+
+    @protected
+    def validate(self):
+        for song in self.songlist:
+            for ch in song:
+                if not isinstance(ch, chord):
+                    raise Exception("Incorrect chord type")
+                    #will also chech cq_max
+
+
+    #returns songlist with Chord class instances

```

Comment:
I don't think so. IMO `load()` should return a list of lists or a map of lists, with the first layer representing a song, and a second list of segments representing sequence of chord with their start and end timestamps.